bookkeeper-server/src/main/java/org/apache/bookkeeper/client/EnsemblePlacementPolicy.java

```diff
@@ -364,6 +364,18 @@ DistributionSchedule.WriteSet reorderReadLACSequence(
     default void updateBookieInfo(Map<BookieId, BookieInfo> bookieInfoMap) {
     }
 
+    /**
+     * Replace some bookie to adhering placement policy. If the all kinds of replacement
+     * didn't adhere placement policy, return empty map.
+     *
+     * @param ensemble
+     * @param writeQuorumSize
+     * @param ackQuorumSize
+     * @return Map: key means ensemble index, value means target replace bookieId.
+     */
+    Map<Integer, BookieId> replaceNotAdheringPlacementPolicyBookie(List<BookieId> ensemble, int writeQuorumSize,
```

Comment:
That's fine.

Reply:
If we cherry-pick this to a minor release, this will break the existing implementation which is out of this repo. So I would suggest using the default to avoid this.

Reply:
yes, we must add a default implementation